src/cli/cli_tcp.cpp

```diff
@@ -463,29 +463,52 @@ template <> otError TcpExample::Process<Cmd("connect")>(Arg aArgs[])
  * tcp send hello
  * Done
  * @endcode
- * @cparam tcp send @ca{message}
+ * @code
+ * tcp send -x 68656c6c6f
+ * Done
+ * @endcode
+ * @cparam tcp send [@ca{type}] @ca{message}
  * The `message` parameter contains the message you want to send to the
  * remote TCP endpoint.
+ * If `type` is `-x`:
+ * Binary data in hexadecimal representation is given in the `message` parameter.
```

Comment:
```suggestion
 * If `type` is `-x`, then
 * binary data in hexadecimal representation is given in the `message` parameter.
```

Line breaks won't be respected on the doc site so better to make it a regular sentence.